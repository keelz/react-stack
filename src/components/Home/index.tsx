import * as React from 'react';

const Home: React.SFC = () => <div className="Home">home</div>;

export default Home;
