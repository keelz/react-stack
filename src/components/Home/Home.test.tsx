import * as React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Home from './index';

const adapter = new Adapter();
configure({ adapter });

describe('Home suite', () => {
  it('should match snapshot', () => {
    const tree = shallow(<Home />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should be selectable by .Home', () => {
    expect(shallow(<Home />).is('.Home')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(<Home />).find('.Home').length).toBe(1);
  });
});
