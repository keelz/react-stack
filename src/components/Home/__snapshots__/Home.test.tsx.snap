// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home suite should match snapshot 1`] = `
<div
  className="Home"
>
  home
</div>
`;
